### PR TITLE
user adds brewpub, admins, and authentication/edit page started

### PR DESCRIPTION
**Pull request by Jennings824**

### CODE REVIEW

File: app/controllers/brewpubs_controller.rb [hunk 1]
```diff
@@ -1,5 +1,55 @@
...
+      redirect_to root_path, notice: "Brewpub added successfully"
+    else
+      flash[:alert] = @brewpub.errors.full_messages.join(',')
+      render :new
+    end
+  end
+
+  def show
+    @brewpub = Brewpub.find(params[:id])
+    # @reviews = @brewpub.reviews
```
Comment (smkopp92) [inline app/controllers/brewpubs_controller.rb]: Try not to leave commented out code in your PR's

File: app/controllers/brewpubs_controller.rb [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+  end
+
+  def edit
+    @brewpub = Brewpub.find(params[:id])
+  end
+
+  def update
+    @brewpub = Brewpub.find(params[:id])
+    # This doesn't save yet
```
Comment (smkopp92) [inline app/controllers/brewpubs_controller.rb]: Be careful leaving comments in your code. You can commit features that aren't finished and put them in a separate branch

Comment (Jennings824) [reply app/controllers/brewpubs_controller.rb]: makes sense

File: spec/features/user_adds_new_brewpub_spec.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
+require "rails_helper"
```
Comment (smkopp92) [inline spec/features/user_adds_new_brewpub_spec.rb]: Generally, you should test all your validations in your model. In your feature, you should test 1 happy path and 1 UNIQUE sad path (happy: user submits information correctly, sad: user submits information incorrectly). It's just much easier to test the granular issues on the model level, and keeping your feature tests simple (Your sad path should test for a flash message, lack of creation, and rerendering of the form)

**Pull request merged**